src/backend/dx12/src/device.rs

```diff
@@ -1318,12 +1354,92 @@ impl d::Device<B> for Device {
     }
 
     fn update_descriptor_sets(&mut self, writes: &[pso::DescriptorSetWrite<B>]) {
+        // Create temporary non-shader visible views for uniform and storage buffers.
+        let mut num_views = 0;
+        for sw in writes {
+            match sw.write {
+                pso::DescriptorWrite::UniformBuffer(ref views) |
+                pso::DescriptorWrite::StorageBuffer(ref views) => {
+                    num_views += views.len();
+                },
+                _ => (),
+            }
+        }
+
+        let mut buffer_heap = n::DescriptorCpuPool {
+            heap: Self::create_descriptor_heap_impl(
```

Comment:
wait, we create a temporary heap upon updating descriptor sets? that sound heavy. Could you raise the topic on an issue/gitter please?